### PR TITLE
Add CVE-2019-18838

### PR DESCRIPTION
**Pull request by fpesce**
The security advisory was missing one vulnerability.

### CODE REVIEW

**Comment by geeknoid** [comment]
@googlebot I consent

File: content/en/news/releases/1.2.x/announcing-1.2.10/index.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ This release contains fixes for the security vulnerability described in [our Dec
 
 __[CVE-2019-18801](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-18801)__: Fix a vulnerability affecting Envoy's processing of large HTTP/2 request headers.  A successful exploitation of this vulnerability could lead to a denial of service, escalation of privileges, or information disclosure.
 __[CVE-2019-18802](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-18802)__: Fix a vulnerability resulting from whitespace after HTTP/1 header values which could allow an attacker to bypass Istio's policy checks, potentially resulting in information disclosure or escalation of privileges.
+__[CVE-2019-18838](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-18838)__: Fix a vulnerability resulting from malformed HTTP request missing the "Host" header. An encoder filter that invokes Envoy's route manager APIs that access request's "Host" header will cause NULL pointer to be dereferenced and result in abnormal termination of the Envoy process.
```
Comment (brian-avery) [inline content/en/news/releases/1.2.x/announcing-1.2.10/index.md]: will cause "a" NULL pointer to be dereferenced.

Comment (fpesce) [inline content/en/news/releases/1.2.x/announcing-1.2.10/index.md]:
```suggestion
__[CVE-2019-18838](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-18838)__: Fix a vulnerability resulting from malformed HTTP request missing the "Host" header. An encoder filter that invokes Envoy's route manager APIs that access request's "Host" header will cause a NULL pointer to be dereferenced and result in abnormal termination of the Envoy process.
```

File: content/en/news/releases/1.3.x/announcing-1.3.6/index.md [hunk 2]
```diff
@@ -19,6 +19,7 @@ This release contains fixes for the security vulnerability described in [our Dec
 
 __[CVE-2019-18801](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-18801)__: Fix a vulnerability affecting Envoy's processing of large HTTP/2 request headers.  A successful exploitation of this vulnerability could lead to a denial of service, escalation of privileges, or information disclosure.
 __[CVE-2019-18802](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-18802)__: Fix a vulnerability resulting from whitespace after HTTP/1 header values which could allow an attacker to bypass Istio's policy checks, potentially resulting in information disclosure or escalation of privileges.
+__[CVE-2019-18838](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-18838)__: Fix a vulnerability resulting from malformed HTTP request missing the "Host" header. An encoder filter that invokes Envoy's route manager APIs that access request's "Host" header will cause NULL pointer to be dereferenced and result in abnormal termination of the Envoy process.
```
Comment (fpesce) [inline content/en/news/releases/1.3.x/announcing-1.3.6/index.md]:
```suggestion
__[CVE-2019-18838](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-18838)__: Fix a vulnerability resulting from malformed HTTP request missing the "Host" header. An encoder filter that invokes Envoy's route manager APIs that access request's "Host" header will cause a NULL pointer to be dereferenced and result in abnormal termination of the Envoy process.
```

File: content/en/news/releases/1.4.x/announcing-1.4.2/index.md [hunk 3]
```diff
@@ -19,3 +19,4 @@ This release contains fixes for the security vulnerability described in [our Dec
 
 __[CVE-2019-18801](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-18801)__: Fix a vulnerability affecting Envoy's processing of large HTTP/2 request headers.  A successful exploitation of this vulnerability could lead to a denial of service, escalation of privileges, or information disclosure.
 __[CVE-2019-18802](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-18802)__: Fix a vulnerability resulting from whitespace after HTTP/1 header values which could allow an attacker to bypass Istio's policy checks, potentially resulting in information disclosure or escalation of privileges.
+__[CVE-2019-18838](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-18838)__: Fix a vulnerability resulting from malformed HTTP request missing the "Host" header. An encoder filter that invokes Envoy's route manager APIs that access request's "Host" header will cause NULL pointer to be dereferenced and result in abnormal termination of the Envoy process.
```
Comment (fpesce) [inline content/en/news/releases/1.4.x/announcing-1.4.2/index.md]:
```suggestion
__[CVE-2019-18838](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-18838)__: Fix a vulnerability resulting from malformed HTTP request missing the "Host" header. An encoder filter that invokes Envoy's route manager APIs that access request's "Host" header will cause a NULL pointer to be dereferenced and result in abnormal termination of the Envoy process.
```

File: content/en/news/security/istio-security-2019-007/index.md [hunk 4]
```diff
@@ -19,14 +19,16 @@ Envoy, and subsequently Istio are vulnerable to two newly discovered vulnerabili
 
 * __[CVE-2019-18802](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-18802)__: HTTP/1 codec incorrectly fails to trim whitespace after header values. This could allow an attacker to bypass Istio's policy either for information disclosure or escalation of privileges.
 
+* __[CVE-2019-18838](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-18838)__: Upon receipt of a malformed HTTP request without the "Host" header, an encoder filter invoking Envoy's route manager APIs that access request's "Host" header will cause NULL pointer to be dereferenced and result in abnormal termination of the Envoy process.
```
Comment (fpesce) [inline content/en/news/security/istio-security-2019-007/index.md]:
```suggestion
* __[CVE-2019-18838](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-18838)__: Upon receipt of a malformed HTTP request without the "Host" header, an encoder filter invoking Envoy's route manager APIs that access request's "Host" header will cause a NULL pointer to be dereferenced and result in abnormal termination of the Envoy process.
```

**Comment by fpesce** [comment]
@googlebot I fixed it.

**Comment by fpesce** [comment]
CLA verified manually. @geeknoid said he consented.

**Pull request merged**